streams/src/test/java/org/apache/kafka/test/NoOpReadOnlyStore.java

```diff
@@ -60,6 +60,11 @@ public V get(final K key) {
         return null;
     }
 
+    @Override
+    public KeyValueIterator<K, V> prefixScan(K prefix) {
+        throw new UnsupportedOperationException(getClass().getSimpleName() + " can't perform a prefix scan");
```

Comment:
this violates LSP, this is not a proper subclass for ReadOnlyKeyValueStore. Better to create a new interface that contains `KeyValueIterator<K, V> prefixScan(K prefix);` and then types (key value stores) that actually support this operation implement this interface.